test/e2e/kubevirt.go

```diff
@@ -721,159 +813,338 @@ passwd:
-	}
+				},
+			}
+			liveMigrate = testCommand{
+				description: "live migration",
+				cmd: func() {
+					liveMigrateVirtualMachine(vmi.Name, kubevirtv1.MigrationPreCopy)
+					checkLiveMigrationSucceeded(vmi.Name, kubevirtv1.MigrationPreCopy)
+				},
+			}
+			butane = `
+variant: fcos
+version: 1.4.0
+passwd:
+  users:
+  - name: core
+    password_hash: $y$j9T$b7RFf2LW7MUOiF4RyLHKA0$T.Ap/uzmg8zrTcUNXyXvBvT26UgkC6zZUVg3UKXeEp5
+`
+			virtualMachine = resourceCommand{
+				description: "VirtualMachine",
+				cmd: func() string {
+					var err error
+					vm, err = fcosVM(1, nil /*labels*/, nil /*annotations*/, nil /*nodeSelector*/, kubevirtv1.NetworkSource{
+						Multus: &kubevirtv1.MultusNetwork{
+							NetworkName: nad.Name,
+						},
+					}, butane)
+					Expect(err).ToNot(HaveOccurred())
+					createVirtualMachine(vm)
+					return vm.Name
+				},
+			}
 
-	var migrations []kubevirtv1.VirtualMachineInstanceMigration
-	for i := range unstructuredVMIMigrations.Items {
-		var vmiMigration kubevirtv1.VirtualMachineInstanceMigration
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(
-			unstructuredVMIMigrations.Items[i].Object,
-			&vmiMigration,
-		); err != nil {
-			return nil, err
+			virtualMachineInstance = resourceCommand{
+				description: "VirtualMachineInstance",
+				cmd: func() string {
+					var err error
+					vmi, err = fcosVMI(1, nil /*labels*/, nil /*annotations*/, nil /*nodeSelector*/, kubevirtv1.NetworkSource{
+						Multus: &kubevirtv1.MultusNetwork{
+							NetworkName: nad.Name,
+						},
+					}, butane)
+					Expect(err).ToNot(HaveOccurred())
+					createVirtualMachineInstance(vmi)
+					return vmi.Name
+				},
+			}
+		)
+		type testData struct {
+			description string
+			resource    resourceCommand
+			test        testCommand
+			topology    string
 		}
-		migrations = append(migrations, vmiMigration)
-	}
+		DescribeTable("should keep ip", func(td testData) {
+			By("Creating NetworkAttachmentDefinition")
+			nad = generateNAD(newNetworkAttachmentConfig(networkAttachmentConfigParams{
+				namespace:          namespace,
+				name:               "net1",
+				topology:           td.topology,
+				cidr:               strings.Join([]string{cidrIPv4, cidrIPv6}, ","),
+				allowPersistentIPs: true,
+			}))
+			Expect(crClient.Create(context.Background(), nad)).To(Succeed())
+
+			vmiName := td.resource.cmd()
+			vmi = &kubevirtv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      vmiName,
+				},
+			}
+			waitVirtualMachineInstanceReadiness(vmi)
+			Expect(crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
 
-	return migrations, nil
-}
+			step := by(vm.Name, "Login to virtual machine for the first time")
+			Expect(kubevirt.LoginToFedora(vmi, "core", "fedora")).To(Succeed(), step)
+			expectedAddreses = waitVirtualMachineAddressesFromStatus(vmi)
+
+			td.test.cmd()
+			time.Sleep(time.Second)
```

Comment:
I don't get this sleep, but it does not look good. No sleep does.

Reply:
So after a restart/live-migration we have to wait a little for the vmi to catch up with new state, we can try to use a `Eventually` with expected NotReady or the like but I don't think is it worth it.